src/eval_cache.rs

```diff
@@ -108,21 +108,19 @@ impl CacheManifest {
 
     /// Checks and inserts the formatter info into the cache.
     /// If the formatter info has changed, invalidate all the old paths.
-    #[must_use]
-    pub fn update_formatters(self, formatters: BTreeMap<FormatterName, Formatter>) -> Self {
-        let mut new_formatters = BTreeMap::new();
-        let mut new_paths = self.matches.clone();
+    pub fn update_formatters(&mut self, formatters: BTreeMap<FormatterName, Formatter>) {
+        let mut old_formatters = std::mem::replace(&mut self.formatters, BTreeMap::new());
         for (name, fmt) in formatters {
-            match load_formatter_info(&fmt) {
+            match load_formatter_info(fmt) {
                 Ok(new_fmt_info) => {
-                    if let Some(old_fmt_info) = self.formatters.get(&name) {
+                    if let Some(old_fmt_info) = old_formatters.remove(&name) {
                         // Invalidate the old paths if the formatter config has changed.
-                        if old_fmt_info != &new_fmt_info {
-                            new_paths.remove(&name);
+                        if old_fmt_info != new_fmt_info {
```

Comment:
This is a small logic change; we're draining values off `old_formatters` so that when the iteration is done, what remains in it is the set difference compared to `formatters`
We remove `&` because `old_fmt_info` went from `&FormatterInfo` to `FormatterInfo`. They now have the same type, so we can compare directly.